### PR TITLE
Add a unique drop chance display to the raids plugin

### PR DESCRIPTION
**Pull request by Rman887**
Gives the option to display your percentage chance of receiving a unique drop at the Chambers of Xeric. This is calculated using the current amount of total points.

### CODE REVIEW

**Comment by kulershk** [comment]
*Raids drop chance caps to 80%
*if you have more than 712,5k points it shows you more than 100% plus 80% is the cap. next points go for next drop chance
*"Gives the option to display your percentage chance of receiving a unique drop" you probably mean team, unless you are doing solo.

File: runelite-client/src/main/java/net/runelite/client/plugins/raids/RaidsPointsOverlay.java [hunk 1]
```diff
@@ -80,6 +83,42 @@ public Dimension render(Graphics2D graphics)
 			.right(String.valueOf(client.getVar(Varbits.RAID_PARTY_SIZE)))
 			.build());
 
+		if (config.uniqueChance())
+		{
+			double chance = Math.min(totalPoints / 7125.0, 80.0);
+			double chance2 = Math.min((totalPoints - 570000.0) / 7125.0, 80.0);
```
Comment (Jasper-ketelaar) [inline runelite-client/src/main/java/net/runelite/client/plugins/raids/RaidsPointsOverlay.java]: No need to calculate chance2/chance3 if the total points aren't even able to get you a drop. Would be better to move this inside their respective conditional blocks.

File: runelite-client/src/main/java/net/runelite/client/plugins/raids/RaidsPointsOverlay.java [hunk 2]
```diff
@@ -80,6 +83,42 @@ public Dimension render(Graphics2D graphics)
...
+			double chance3 = Math.min((totalPoints - 1140000.0) / 7125.0, 80.0);
+
+			this.panel.setPreferredSize(new Dimension(ComponentConstants.STANDARD_WIDTH + 10, 0));
+			panel.getChildren().add(LineComponent.builder()
+				.left("Unique Chance:")
+				.right(String.valueOf(Math.round(chance * 10.0) / 10.0) + "%")
```
Comment (Jasper-ketelaar) [inline runelite-client/src/main/java/net/runelite/client/plugins/raids/RaidsPointsOverlay.java]: Use DecimalFormat

File: runelite-client/src/main/java/net/runelite/client/plugins/raids/RaidsPointsOverlay.java [hunk 3]
```diff
@@ -80,6 +83,42 @@ public Dimension render(Graphics2D graphics)
...
+				if (totalPoints >= 1140000.0)
+				{
+					panel.getChildren().add(LineComponent.builder()
+						.left("Unique Chance 3:")
+						.right(String.valueOf(Math.round(chance3 * 10.0) / 10.0) + "%")
+						.build());
+				}
+			}
+
+
```
Comment (Jasper-ketelaar) [inline runelite-client/src/main/java/net/runelite/client/plugins/raids/RaidsPointsOverlay.java]: Remove empty line

File: runelite-client/src/main/java/net/runelite/client/plugins/raids/RaidsPointsOverlay.java [hunk 4]
```diff
@@ -80,6 +83,42 @@ public Dimension render(Graphics2D graphics)
...
+				.left("Unique Chance:")
+				.right(String.valueOf(Math.round(chance * 10.0) / 10.0) + "%")
+				.build());
+
+			if (totalPoints >= 570000.0)
+			{
+				this.panel.setPreferredSize(new Dimension(ComponentConstants.STANDARD_WIDTH + 20, 0));
+				panel.getChildren().add(LineComponent.builder()
+					.left("Unique Chance 2:")
+					.right(String.valueOf(Math.round(chance2 * 10.0) / 10.0) + "%")
```
Comment (Jasper-ketelaar) [inline runelite-client/src/main/java/net/runelite/client/plugins/raids/RaidsPointsOverlay.java]: Use same DecimalFormat

File: runelite-client/src/main/java/net/runelite/client/plugins/raids/RaidsPointsOverlay.java [hunk 5]
```diff
@@ -80,6 +83,42 @@ public Dimension render(Graphics2D graphics)
...
+					.build());
+
+				if (totalPoints >= 1140000.0)
+				{
+					panel.getChildren().add(LineComponent.builder()
+						.left("Unique Chance 3:")
+						.right(String.valueOf(Math.round(chance3 * 10.0) / 10.0) + "%")
```
Comment (Jasper-ketelaar) [inline runelite-client/src/main/java/net/runelite/client/plugins/raids/RaidsPointsOverlay.java]: Decimalformat again

File: runelite-client/src/main/java/net/runelite/client/plugins/raids/RaidsPointsOverlay.java [hunk 6]
```diff
@@ -80,6 +83,42 @@ public Dimension render(Graphics2D graphics)
 			.right(String.valueOf(client.getVar(Varbits.RAID_PARTY_SIZE)))
 			.build());
 
+		if (config.uniqueChance())
+		{
+			double chance = Math.min(totalPoints / 7125.0, 80.0);
```
Comment (Jasper-ketelaar) [inline runelite-client/src/main/java/net/runelite/client/plugins/raids/RaidsPointsOverlay.java]: Just calculate double percent = totalPoints / 7125.0 once and replace 570000.0 with 80 and 1140000.0 with 160.

Comment (Jasper-ketelaar) [reply runelite-client/src/main/java/net/runelite/client/plugins/raids/RaidsPointsOverlay.java]: Also put the double values such as 7125.0 and 80.0 into constants to make sense of their meaning.

Comment (Jasper-ketelaar) [reply runelite-client/src/main/java/net/runelite/client/plugins/raids/RaidsPointsOverlay.java]: Yeah exactly, that would look a lot cleaner. Just make sure not to have any magic numbers.

**Comment by Trevor159** [comment]
We now use the vanilla points overlay. The old RL points overlay was moved to an external plugin. If you PR this to the external plugin I will accept it.

**Pull request closed without merging**